ballerina/tests/xml_from_json_test.bal

```diff
@@ -137,46 +101,380 @@ isolated function testAttributeValidation() {
+                    "<bookId>102</bookId>" +
+                "</item>" +
+            "</item>" +
+            "<item>" +
+                "<item>" +
+                    "<bookName>book3</bookName>" +
+                    "<bookId>103</bookId>" +
+                "</item>" +
+            "</item>" +
+        "</books>" +
+    "</root>";
+    xml?|error result = fromJson(data);
+    if result is xml {
+        test:assertEquals(result.toString(), expected);
+    } else {
+        test:assertFail("failed to convert json to xml");
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testArray() {
+    json data = [
+        {
+            fname: "foo",
+            lname: "bar"
+        },
+        1
+    ];
+    string expected =
+    "<root>" +
+        "<item>" +
+            "<fname>foo</fname>" +
+            "<lname>bar</lname>" +
+        "</item>" +
+        "<item>1</item>" +
+    "</root>";
+    xml?|error result = fromJson(data);
+    if result is xml {
+        test:assertEquals(result.toString(), expected);
+    } else {
+        test:assertFail("failed to convert json to xml");
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testWithAttribute() {
+    json data = {
+        "@series": "Dark",
+        genre: "Sci-Fi",
+        language: "German",
+        seasons: 3,
+        "@id": 3296
+    };
+    xml?|error result = fromJson(data);
+    string expected =
+    "<root series=\"Dark\" id=\"3296\">" +
+        "<genre>Sci-Fi</genre>" +
+        "<language>German</language>" +
+        "<seasons>3</seasons>" +
+    "</root>";
+    if result is xml {
+        test:assertEquals(result.toString(), expected);
+    } else {
+        test:assertFail("failed to convert json to xml");
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testNamespace() {
+    json data = {
+        "ns0:bookStore": {
+            "@xmlns:ns0": "http://sample.com/test",
+            "@status": "online",
+            "ns0:storeName": "foo",
+            "ns0:postalCode": "94",
+            "ns0:isOpen": "true",
+            "ns0:address": {
+                "ns0:street": "No 20, Palm Grove",
+                "ns0:city": "Colombo 03",
+                "ns0:country": "Sri Lanka"
+            },
+            "ns0:codes": ["4", "8", "9"]
+        },
+        "metaInfo": "some info"
+    };
+    string expected =
+    "<root>" +
+        "<ns0:bookStore xmlns:ns0=\"http://sample.com/test\" status=\"online\">" +
+            "<ns0:storeName>foo</ns0:storeName>" +
+            "<ns0:postalCode>94</ns0:postalCode>" +
+            "<ns0:isOpen>true</ns0:isOpen>" +
+            "<ns0:address>" +
+                "<ns0:street>No 20, Palm Grove</ns0:street>" +
+                "<ns0:city>Colombo 03</ns0:city>" +
+                "<ns0:country>Sri Lanka</ns0:country>" +
+            "</ns0:address>" +
+            "<ns0:codes>" +
+                "<item>4</item>" +
```

Comment:
Don't we add a name prefix?
```suggestion
                "<ns0:item>4</ns0:item>" +
```

Reply:
According to [this](https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html), the latest `toJson()` will not be eliminated the identical names tags, when converting to JSON array objects. Instead of this, that identical name is a JSON array key. If this XML is used as input for toJson(), the output is: 
https://docs.google.com/document/d/1m_IJgE3qJjENauWCiCTUxubB9pg2mH3ye-i6-HwNUjU/edit#bookmark=id.cmiytjxkma5x

So IMO, we don't need to create the new tag name as `item` in the XML. Instead of this, we can create a XML with identical key following way: `<ns0:codes>4</ns0:codes><ns0:codes>8</ns0:codes><ns0:codes>9</ns0:codes>`

@daneshk & @MadhukaHarith92 WDYT?

Reply:
According to the spec[[1]](https://docs.google.com/document/d/1m_IJgE3qJjENauWCiCTUxubB9pg2mH3ye-i6-HwNUjU/edit#bookmark=id.1fn2rkgycu7h), the array element is created inside the `item` tag.

[1] https://docs.google.com/document/d/1m_IJgE3qJjENauWCiCTUxubB9pg2mH3ye-i6-HwNUjU/edit#bookmark=id.1fn2rkgycu7h

Reply:
> 

I think the prefix is optional for an array element tag. Checked a couple of sources and in some cases, the prefix is there, and in other places, the prefix isn't there.

Reply:
> 

Yes, but the`fromJson` creates an array with the tag as `item` default. So I think this is not good. If the user wants an array tag, the user needs to give the value. Otherwise, we don't need to create a new array tag as `item`.

Reply:
Ack. Could you create an issue since this requires a change in the spec? I think there are pros and cons to the suggested approach.

Reply:
IMO, If the user doesn't specifically mention abt the `arrayEntryTag`, we should repeat the JSON key, so it does not break the reverse conversion from XML to JSON

Reply:
We need to create a spec for XML data and correct this.